searx/engines/solidtorrents.py

```diff
@@ -1,51 +1,85 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # lint: pylint
-"""Solid Torrents
-
+"""SolidTorrents
 """
 
-from json import loads
+from datetime import datetime
 from urllib.parse import urlencode
+import random
+
+from lxml import html
+
+from searx.utils import extract_text, eval_xpath, eval_xpath_getindex
 
 about = {
     "website": 'https://www.solidtorrents.net/',
     "wikidata_id": None,
     "official_api_documentation": None,
-    "use_official_api": True,
+    "use_official_api": False,
     "require_api_key": False,
-    "results": 'JSON',
+    "results": 'HTML',
 }
 
 categories = ['files']
 paging = True
 
-base_url = 'https://www.solidtorrents.net/'
-search_url = base_url + 'api/v1/search?{query}'
+base_url = ''
+base_url_rand = ''
+
+units = {"B": 1, "KB": 2 ** 10, "MB": 2 ** 20, "GB": 2 ** 30, "TB": 2 ** 40}
```

Comment:
Note: [`get_torrent_size`](https://github.com/searxng/searxng/blob/de32d543bc5e7fac968a11c7cdd0e3b8f349a94d/searx/utils.py#L263-L301) in `searx.utils` convert the file size to bytes. The current implementation of this function misses the `B` unit.

Reply:
Thanks for pointing out [get_torrent_size](https://github.com/searxng/searxng/blob/de32d543bc5e7fac968a11c7cdd0e3b8f349a94d/searx/utils.py#L263-L301) I haven't seen / fixed.